src/SpendShelf.BankTransactionsImport/TransactionsUpload/BankFileInfo.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System.ComponentModel.DataAnnotations;
+using Microsoft.AspNetCore.Http;
+using SpendShelf.BankTransactionsImport.Infrastructure.Attributes;
+
+namespace SpendShelf.BankTransactionsImport.TransactionsUpload
+{
+    public class BankFileInfo
+    {
+        [Required(ErrorMessage = "Please select a file.")]
+        [DataType(DataType.Upload)]
+        [MaxFileSize(10 * 1024 * 1024)]
+        [AllowedExtensions(new string[] { ".xls" })]
```

Comment:
If you want to limit filetype extension is not a reliable way to achieve this.
It's better to parse file metadata

Reply:
It's not required in this specific example through
Just a suggestion

Reply:
I'm totally agree with you it's the most reliable way to check file extension. But it's good enough to check if request was valid or not. I think if it will be needed and possible we can add additional metadata validation in the future. What to you think?

Reply:
yes